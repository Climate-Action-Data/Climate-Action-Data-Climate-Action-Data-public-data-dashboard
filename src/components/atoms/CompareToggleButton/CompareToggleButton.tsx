import { Button } from '@chakra-ui/react'
import { FC } from 'react'

export interface CompareToggleButtonProps {
  isEnabled?: boolean
  onClick: () => void
}

const CompareToggleButton: FC<CompareToggleButtonProps> = (props) => {
  const { onClick } = props
  return (
    <Button padding="6px 8px 6px 12p" width="max-content" onClick={onClick} borderRadius="32px" fontSize="14px" fontWeight="medium" data-testid="compare-toggle-button">
      Compare
    </Button>
  )
}

export default CompareToggleButton
