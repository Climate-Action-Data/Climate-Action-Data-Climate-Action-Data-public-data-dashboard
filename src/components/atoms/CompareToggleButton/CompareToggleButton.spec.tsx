import { render } from '@testing-library/react'
import CompareToggleButton, { CompareToggleButtonProps } from './CompareToggleButton'

const DEFAULT_PROPS: CompareToggleButtonProps = {
  isEnabled: true,
  onClick: () => jest.fn(),
}

it(`renders correctly`, () => {
  const { container } = render(<CompareToggleButton onClick={DEFAULT_PROPS.onClick} isEnabled={DEFAULT_PROPS.isEnabled} />)
  expect(container).toMatchSnapshot()
})
