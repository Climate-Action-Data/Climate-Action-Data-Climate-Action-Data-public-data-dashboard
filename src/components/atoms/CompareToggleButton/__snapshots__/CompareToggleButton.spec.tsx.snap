// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    class="chakra-button css-eeegjl"
    data-testid="compare-toggle-button"
    type="button"
  >
    Compare
  </button>
</div>
`;
